modules/Emails/EmailsController.php

```diff
@@ -813,12 +814,14 @@ protected function userIsAllowedToSendEmail($requestedUser, $requestedInboundEma
 
         $inboundEmailStoredOptions = $requestedInboundEmail->getStoredOptions();
 
-        // if group email account, check that user is allowed to use group email account
-        if ($requestedInboundEmail->isGroupEmailAccount()) {
-            if ($inboundEmailStoredOptions['allow_outbound_group_usage'] === true) {
-                $hasAccessToInboundEmailAccount = true;
-            } else {
-                $hasAccessToInboundEmailAccount = false;
+        if (!$hasAccessToInboundEmailAccount) {
```

Comment:
The $requestedInboundEmail could be anything and the retrieveAllByGroupIdWithGroupAccounts should return only the non-personal accounts. Btw you'r right, this part logically removable if isGroupEmailAccount() will works properly.